src/common/utility_unix.cpp

```diff
@@ -65,12 +65,27 @@ void setLaunchOnStartup_private(const QString &appName, const QString &guiName,
             qCWarning(lcUtility) << "Could not write auto start entry" << desktopFileLocation;
             return;
         }
+
+        auto autostartApplicationPath = []() {
+            // $APPIMAGE will be set to the AppImage's path by the AppImage runtime
+            // if it is set, we can assume to be run from within an AppImage
+            // in that case, the desktop file should point to the AppImage rather than the
+            // main binary, which will be in a temporary mount point
+            static const char APPIMAGE[] = "APPIMAGE";
```

Comment:
As that env var is now used and defined in multiple places, maybe add it to Utility?

Reply:
I was thinking about something like this:

```cpp
std::optional<QString> appImageLocation();
```

Qt doesn't seem to have an equivalent of `std::optional`, `QVariant` could be used. I guess we can also define an empty string as "not running in AppImage".

Reply:
```c++
QString appImageLocation(){
static QString loc = qGetEnv();
retun loc;
}
bool isAppimage() {
   return !appImageLocation().isEmpty();
}
```

I don't think std::optional would add much here.